translate newsletter286 into zh (#94)
PrimitivesLane/bitcoinops.github.io: _posts/zh/newsletters/2024-01-24-newsletter.md

---
title: 'Bitcoin Optech Newsletter #286'
permalink: /zh/newsletters/2024/01/24/
name: 2024-01-24-newsletter-zh
slug: 2024-01-24-newsletter-zh
type: newsletter
layout: newsletter
lang: zh
---

本周的周报介绍了旧版 btcd 软件中的一个已经修复的共识故障、为闪电网络使用 “v3 交易转发” 和 “一次性锚点” 而提出的变更，以及比特币相关规范的一个新仓库。此外是我们的常规栏目：服务和客户端软件的升级介绍、软件新版本和候选版本的介绍，以及热门比特币基础设施软件的重大变更总结。

## 新闻

> - **<!--disclosure-of-fixed-consensus-failure-in-btcd-->已修复的 btcd 共识故障披露**：Niklas Gögge 在 Delving Bitcoin 论坛[公开][gogge btcd]了他已经[尽责披露][topic responsible disclosures]的旧版 btcd 软件中的一个共识故障。相对[时间锁][topic timelocks]是通过向交易输入的 sequence 字段添加共识强制的含义的软分叉而[添加到比特币中的][topic soft fork activation]。为了保证在分叉之前创建的任何预签名交易都不会因此而作废，相对时间锁仅对交易版本号为 2 乃至更高数字的数字，以允许原本默认版本号为 1 的交易可以容许任意输入。然而，在最初的比特币软件中，版本 号是有符号的整数，意思是负数也是有可能的。而 [BIP68][] 的 “参考实现” 部分指出，“版本号为 2 以及更高” 意味着将版本号从有符号的整数[转变][cast]为无符号的整数，从而保证这个规则适用于任何交易版本号不为 0 和 1 的交易。

Reviewer: ```suggestion
- **<!--disclosure-of-fixed-consensus-failure-in-btcd-->已修复的 btcd 共识故障披露**：Niklas Gögge 在 Delving Bitcoin 论坛[公开][gogge btcd]了他已经[尽责披露][topic responsible disclosures]的旧版 btcd 软件中的一个共识故障。相对[时间锁][topic timelocks]是通过向交易输入的 sequence 字段添加共识强制的含义的软分叉而[添加到比特币中的][topic soft fork activation]。为了保证在分叉之前创建的任何预签名交易都不会因此而作废，相对时间锁仅对交易版本号为 2 乃至更高数字的数字，以允许原本默认版本号为 1 的交易可以容许任意输入。然而，在最初的比特币软件中，版本号是有符号的整数，即也可能是负数。而 [BIP68][] 的 “参考实现” 部分指出，“版本号为 2 以及更高” 意味着将版本号从有符号的整数[转变][cast]为无符号的整数，从而保证这个规则适用于任何交易版本号不为 0 和 1 的交易。
```

  - *<!--anchor-swap-->锚点切换*：当前为了在 “[CPFP carve out][topic cpfp carve out]” 规则下保证 [CPFP 手续费追加][topic cpfp] 能力而放置在承诺交易中的两个[锚点输出][topic anchor outputs]需要移除，代之以一个一次性锚点输出。

  - *<!--reducing-delays-->减少延迟*：承诺交易输出的额外 1 个区块的时延需要移除。使用它们是为了保证 CPFP carve out 总是能工作，但在 v3 交易转发规则下已不再必要。

  - *<!--trimming-redirect-->隐式 HTLC 重定向* ：不再将所有 “[隐式 HTLC][topic trimmed htlc]” 的价值放到承诺交易的手续费中，而将它们的总价值加到锚点输出的价值中，以保证额外的手续费会被用来保证承诺交易以及一次性锚点的确认（见[上周的周报][news285 mev]的讨论）。

  - *<!--other-changes-->其它变更*：其它一些较小的变更以及简化。

  后续的讨论检查了这些变更的一些有趣的后果，包括：

  - *<!--reduced-utxo-requirements-->减少 UTXO 需求*：得益于额外 1 区块时延的移除，保证正确的通道状态上链变得更加简单。如果欺诈的一方广播了一笔已经撤销的承诺交易，另一方可以立即使用该承诺交易中属于自己的主要输出、以 CPFP 法增加该承诺交易的手续费。他们不再需要保留一个单独的、已经确认的 UTXO 以备 CPFP 使用。为了确保安全，通道的参与者应该保留足够的储备余额，因为 [BOLT2][] 中指定的 1% 下限可能不足以覆盖手续费。对于保留了足够多储备的非转发节点，出于安全理由而需要额外 UTXO 的唯一场景只剩他们要接收一笔进入的支付的时候。

  - *<!--imbued-v3-logic-->渗透 v3 逻辑*：在回应闪电网络规范会议中出现的担忧（闪电网络可能需要很长时间来设计、实现和部署这些变更）时，Gregory Snders [提议][sanders transition]采用一个过渡阶段，对看起来像当前锚点风格的闪电网络承诺交易进行临时的特殊处理，以允许 Bitcoin Core 部署族群交易池而不会被闪电网络的开发阻碍。当软件已经得到广泛部署、所有的闪电客户端实现都升级到使用 v3 时，可以抛弃过渡的特殊规则。

>   - *<!--request-for-max-child-size-->请求为子交易体积设置上限*：v3 交易转发的提议草案将待确认交易的子交易的体积上限设为 1000 vbyte。这个体积越大，承诺交易需要为克服 “[交易钉死][topic transaction pinning]” 而支付的手续费就越多（见[周报 #283][news283 v3pin]）。这个体积越小，诚实用户可以用来贡献手续费的输入位置就越少。

Reviewer: ```suggestion
  - *<!--request-for-max-child-size-->请求为子交易体积设置上限*：v3 交易转发的提议草案将待确认交易的子交易的体积上限设为 1000 vbyte。这个体积越大，诚实用户需要为克服 “[交易钉死][topic transaction pinning]” 而支付的手续费就越多（见[周报 #283][news283 v3pin]）。这个体积越小，诚实用户可以用来贡献手续费的输入位置就越少。
```

- **<!--new-documentation-repository-->新的文档仓库**：Anthony Towns 在 Bitcoin-Dev 邮件组中[公布][towns binana]了一个用于存放协议规范的新仓库：*Bitcoin Inquisition Numbers And Names Authority*（[BINANA][binana repo]）。截至本刊撰写之时，仓库内已有 4 个规范：

    - [BIN24-1][] `OP_CAT`，由 Ethan Heilman 和 Armin Sabouri 撰写。[周报 #274][news274 cat] 有他们的软分叉提议的介绍。

    - [BIN24-2][] Heretical Deployments，由 Anthony Towns 撰写，介绍了 [Bitcoin Inquisition][bitcoin inquisition repo] 软件在默认的 [signet][topic signet] 上测试软分叉提议和其它变更的用法。详细介绍见[周报 #232][news232 inqui]。

    - [BIN24-3][]  `OP_CHECKSIGFROMSTACK`，由 Brandon Black 撰写，说明了这个[提出了很长时间的想法][topic OP_CHECKSIGFROMSTACK]。[上周的周报][news285 lnhance]介绍了 Black 将该操作码放进 LNHANCE 软分叉组合的的提议。

>     - [BIN24-4][]  `OP_INTERNALKEY`，由 Brandon Black 撰写，说明了一个从脚本解释器种检索 taproot 内部公钥的操作码。它也是上周介绍的 LNHANCE 软分叉组合的一部分。

Reviewer: ```suggestion
    - [BIN24-4][]  `OP_INTERNALKEY`，由 Brandon Black 撰写，说明了一个从脚本解释器中检索 taproot 内部公钥的操作码。它也是上周介绍的 LNHANCE 软分叉组合的一部分。
```

    Bitcoin Optech 已经将 BINANA 仓库加入我们观察更新的文档资源清单，该清单也包括 BIP、BOLT 和 BLIP。未来的更新可能会出现在 “重大代码和文档变更” 栏目中。

## 服务和客户端软件的变更

*在这个月度栏目中，我们列出比特币钱包和服务的有趣更新。*

- **<!--envoy-15-released-->Envoy 1.5 推出**：[Envoy 1.5][]增加了对 [taproot][topic taproot] 花费和接收的支持，并改变了处理[不经济的输出][topic uneconomical outputs]的方法；此外还有一些 bug 修复和[其它更新][envoy blog]。

> - **<!--liana-v40-released-->Liana v4.0 推出**：[Liana v4.0][] 已经 [released][liana blog]，包含了对 [RBF 手续费追加法][topic rbf]、使用 RBF 取消交易、自动化[找钱][topic coin selection]以及硬件签名设备地址验证的支持。

Reviewer: coin selection ： 选币？
